### PR TITLE
Compile unf_ext gem from source

### PR DESCRIPTION
**Pull request by johnmccrae**
## Description
unf_ext 0.0.8.2 does not compile on Windows when loaded from rubygems. JSnapp found a workaround by compiling from source. We add that below and removed the dependency on 0.0.8.1 we added into Gemfile. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).
Pre-compiled unf_ext gem in rubygems.org has broken ruby version
constraints which prevent it from installing in ruby 3.1.1 on 64 bit
Windows so we need to install it from source until the issue is resolved.
reference: knu/ruby-unf_ext#66

Signed-off-by: Jeremiah Snapp <jeremiah@chef.io>

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -39,7 +47,7 @@ gem "proxifier", git: "https://github.com/chef/ruby-proxifier", branch: "lcg/rub
 # Everything except AIX and Windows
 group(:ruby_shadow) do
   # if ruby-shadow does a release that supports ruby-3.0 this can be removed
-  gem "ruby-shadow", git: "https://github.com/chef/ruby-shadow", branch: "lcg/ruby-3.0", platforms: :ruby
+  gem "ruby-shadow", git: "https://github.com/chef/ruby-shadow", branch: "lcg/ruby-3.0", platforms: :ruby unless RUBY_PLATFORM.match?(/x64-mingw-ucrt/i)
```
Comment (marcparadise) [inline Gemfile:50]: We'll probably need to use an expression that catches both ruby 3.1 (x64-mingw-ucrt) and ruby 3.0 (x64-mingw32).

File: omnibus_overrides.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@
 override "nokogiri", version: "1.13.1"
 override "openssl", version: mac_os_x? ? "1.1.1m" : "1.0.2zb"
 override "pkg-config-lite", version: "0.28-1"
-override "ruby", version: "3.0.3"
+override "ruby", version: aix? ? "3.0.3" : "3.1.2"
```
Comment (marcparadise) [inline omnibus_overrides.rb:23]: This seems outside the scope of what this PR is intended to so - it will change the omnibus build to use ruby 3.1, but I don't think all of our gems are compatible yet.

**Review by marcparadise — CHANGES_REQUESTED** [review]

**Comment by marcparadise** [comment]
Let's also squash these commits

File: omnibus_overrides.rb (hunk 2 above)
Comment (johnmccrae) [reply omnibus_overrides.rb:23]: undone, good catch.

File: Gemfile (hunk 1 above)
Comment (johnmccrae) [reply Gemfile:50]: Done

**Comment by johnmccrae** [comment]
This is no longer relevant as the gem has updated and chef is compiling without this update
